python/ray/autoscaler/_private/fake_multi_node/node_provider.py

```diff
@@ -1,14 +1,24 @@
+import copy
+import json
 import logging
 import os
-import json
+import subprocess
 from threading import RLock
+import time
+from types import ModuleType
+from typing import Any, Dict, Optional
+import yaml
 
 import ray
+from ray.autoscaler._private.fake_multi_node.command_runner import \
+    FakeDockerCommandRunner
+from ray.autoscaler.command_runner import CommandRunnerInterface
 from ray.autoscaler.node_provider import NodeProvider
 from ray.autoscaler.tags import (TAG_RAY_NODE_KIND, NODE_KIND_HEAD,
                                  NODE_KIND_WORKER, TAG_RAY_USER_NODE_TYPE,
                                  TAG_RAY_NODE_NAME, TAG_RAY_NODE_STATUS,
                                  STATUS_UP_TO_DATE)
+from ray.ray_constants import DEFAULT_PORT
```

Comment:
consider refactoring the docker stuff into a separate class and then documenting the design in file

Reply:
We had that initially but because of shared code moved this into the same class after initial feedback.

If we separate into two classes we will end up with two different node providers, is this something we'd want? I'm happy either way but getting conflicting feedback here. Also cc @ericl for your opinion.

In my opinion, the two node providers serve different purposes, so I'd be happy to separate this into two classes. We could still have a class hierarchy so that the docker compose provider inherits from the fake multi node provider, and keep the code in the same module (but a different file). What are your thoughts?

I'll prepare this after addressing the other feedback.

Reply:
OK given that my comment is later i'd propose to stick to what it is currently.  Sorry about the confusion, thanks for surfacing.